File: BankingAPI/Models/Account.cs
using MongoDB.Bson.Serialization.Attributes;

namespace BankingAPI.Models
{
    [BsonIgnoreExtraElements]
    // Step 2: Define Data Models: Account Model: Should include fields like Id, AccountNumber, AccountType (Savings,Checking),and Balance.
    public class Account
    {
        public int Id { get; set; }

        public int CustomerId { get; set; }

        public string AccountNumber { get; set; } = string.Empty;

        public string AccountType { get; set; } = string.Empty;

        public decimal Balance { get; set; }
    }
}

File: client/src/models/Account.js
export default class Account {
  constructor({
    id = '',
    customerId = '',
    accountNumber = '',
    accountType = '',
    balance = 0,
    customerName = '',
  } = {}) {
    this.id = id
    this.customerId = customerId
    this.accountNumber = accountNumber
    this.accountType = accountType
    this.balance = balance
    this.customerName = customerName
  }

  static from(obj) {
    return new Account({
      id: obj.id,
      customerId: obj.customerId,
      accountNumber: obj.accountNumber,
      accountType: obj.accountType,
      balance: obj.balance,
      customerName: obj.customerName || '',
    })
  }
}